src/Microsoft.Extensions.CodeGenerators.Mvc/View/ViewGenerator.cs

```diff
@@ -62,77 +64,72 @@ public virtual IEnumerable<string> TemplateFolders
-                ? Path.Combine(appbasePath)
-                : Path.Combine(appbasePath, viewGeneratorModel.RelativeFolderPath);
+            var outputPath = ValidateAndGetOutputPath(viewGeneratorModel, outputFileName: viewGeneratorModel.ViewName + Constants.ViewExtension);
+            var modelTypeAndContextModel = await ValidateModelAndGetMetadata(viewGeneratorModel);
 
-            var outputPath = Path.Combine(outputFolder, viewGeneratorModel.ViewName + Constants.ViewExtension);
+            var layoutDependencyInstaller = ActivatorUtilities.CreateInstance<MvcLayoutDependencyInstaller>(_serviceProvider);
+            await layoutDependencyInstaller.Execute();
 
-            if (File.Exists(outputPath) && !viewGeneratorModel.Force)
+            if (viewGeneratorModel.ViewName.EndsWith(Constants.ViewExtension, StringComparison.OrdinalIgnoreCase))
             {
-                throw new InvalidOperationException(string.Format(
-                    CultureInfo.CurrentCulture,
-                    "The file {0} exists, use -f option to overwrite",
-                    outputPath));
+                int viewNameLength = viewGeneratorModel.ViewName.Length - Constants.ViewExtension.Length;
+                viewGeneratorModel.ViewName = viewGeneratorModel.ViewName.Substring(0, viewNameLength);
             }
 
-            var templateName = viewGeneratorModel.TemplateName + Constants.RazorTemplateExtension;
-
-            var dbContextFullName = dataContext != null ? dataContext.FullName : viewGeneratorModel.DataContextClass;
-            var modelTypeFullName = model.FullName;
-
-            var modelMetadata = await _entityFrameworkService.GetModelMetadata(
-                dbContextFullName,
-                model);
-
-            var layoutDependencyInstaller = ActivatorUtilities.CreateInstance<MvcLayoutDependencyInstaller>(_serviceProvider);
-
             bool isLayoutSelected = !viewGeneratorModel.PartialView &&
                 (viewGeneratorModel.UseDefaultLayout || !String.IsNullOrEmpty(viewGeneratorModel.LayoutPage));
 
-            await layoutDependencyInstaller.Execute();
-
             var templateModel = new ViewGeneratorTemplateModel()
             {
-                ViewDataTypeName = modelTypeFullName,
-                ViewDataTypeShortName = model.Name,
+                ViewDataTypeName = modelTypeAndContextModel.ModelType.FullName,
+                ViewDataTypeShortName = modelTypeAndContextModel.ModelType.Name,
                 ViewName = viewGeneratorModel.ViewName,
                 LayoutPageFile = viewGeneratorModel.LayoutPage,
                 IsLayoutPageSelected = isLayoutSelected,
                 IsPartialView = viewGeneratorModel.PartialView,
                 ReferenceScriptLibraries = viewGeneratorModel.ReferenceScriptLibraries,
-                ModelMetadata = modelMetadata,
+                ModelMetadata = modelTypeAndContextModel.ModelMetadata,
                 JQueryVersion = "1.10.2" //Todo
```

Comment:
You mean the JQueryVersion? Yeah, the scripts issues that I logged should be used to fix this...As of now, I think the parameter is not even used in templates ... we need to decide what scripts do we want to reference and what version and accordingly fix this...